File: year2020/month10/day1014/ms_1014.cs
using System;

namespace algorithm_78
{
    class ms_1014
    {
        static void Print(int n, int level) 
        {
            if(n == 1) 
            {
                //Console.WriteLine(n);
                Console.WriteLine(level);
                return;
            } 
            //else Console.Write($"{n} -> ");

            if(n%2 == 0)
                Print(n/2, ++level);
            else
                Print(n/3, ++level);
        }

        static void Main(string[] args)
        {
            int n = Convert.ToInt32(Console.ReadLine());

            if(0 < n && n <= 100) 
                Print(n, 0);
        }
    }
}

File: year2020/month10/day1014/ms_1014.cpp
#include <iostream>

using namespace std;

void print(int n, int level);

int main() {
	int n;
	
	cin >> n;

	if (0 < n && n <= 100)
		print(n, 0);

	return 0;
}

void print(int n, int level) {
	if (n == 1) {
		cout << n << endl;
		cout << level << endl;
		return;
	}
	else cout << n << " -> ";

	if (n%2 == 0)
		print(n / 2, ++level);
	else
		print(n / 3, ++level);
}